### PR TITLE
Moves settings to subdirectory

### PR DESCRIPTION
**Pull request by jswright61**
Settings file moves from home directory/config to config under this root directory.

Modifies README to reflect change.

Adds Advanced Setup section to README

.gigignore moves from heredoc to its own file in config/

Adds variable for LOGFILE_DIR

Nit picky change: tense of verb changed in commit message that adds .gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1 @@
-.DS_Store
-.DS_Store?
-.Spotlight-V100
-.Trashes
-Icon
-Icon*
+config/settings
```
Comment (jswright61) [inline .gitignore:1]: I addressed this in the README, but IMHO these types of files should be ignored globally - it's basically garbage that *your* OS creates, the project should not be cluttered with this.

File: README.md [hunk 2]
```diff
@@ -173,6 +176,29 @@ But remember you can always override the default by using either `--public` or `
 
 Ask away. You can find me [@tjluoma](http://twitter.com/tjluoma) on Twitter, or create an issue on [Bitbucket](https://bitbucket.org/tjluoma/new-dual-repo/issues/new) or [GitHub](https://github.com/tjluoma/new-dual-repo/issues/new).
 
+## Advanced Setup
+
+The following is recommended for advanced users. If something looks too complicated or you don't understand it, maybe
+stick with the [Basic Setup](#basic-setup) above.
+
+Copy config/settings.example to config/settings and modify that file with your individual values. This leaves the new-dual-repo.sh file unchanged and puts all your settings in an untracked file.
+
```
Comment (jswright61) [inline README.md:185]: It bothered me that a file was needed in ~/.config/ and that the file was named github-and-bitbucket.txt. I get that it had generic credentials that could be shared across projects, but to my way of thinking, project settings belong with the project. That said - it's *your* project, so if you like your way, by all means keep your way.

File: config/settings.example [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  # Notes:
+  # 1. you must have a paid GitHub account to create private repos
+  #
+  # 2. if you want to make your repos private by default, uncomment PRIVATE='true' AND
+  #    comment out PRIVATE='false' below
+
+  # PRIVATE='true'
+  PRIVATE='false'
+
+  LOGFILE_DIR="$HOME/Desktop"
```
Comment (jswright61) [inline config/settings.example:25]: I'm a weird Mac guy - I'm not a fan of putting files on my desktop - so I made a variable to give me (and others) options

File: new-dual-repo.sh [hunk 4]
```diff
@@ -288,24 +295,21 @@ git commit -m "Added bare README"	|| die "git commit failed"
...
-Icon
-Icon*
-EOINPUT
-
+	# create a .gitignore
+	if [[ -e "$ROOT_DIR/config/gitignore" ]]
+	then
+		gitignore=$(<"$ROOT_DIR/config/gitignore")
+		echo "$gitignore" > .gitignore
+	else
```
Comment (jswright61) [inline new-dual-repo.sh:305]: I think it's more intuitive and less prone to mistakes to edit a template file than to edit heredocs.

File: new-dual-repo.sh [hunk 5]
```diff
@@ -288,24 +295,21 @@ git commit -m "Added bare README"	|| die "git commit failed"
...
+		echo "# ignore these files (1 file or pattern per line)\n" > .gitignore
+	fi
 fi
 
 [[ -s .gitignore ]] || die "Failed to create .gitignore"
 
 git add .gitignore || die "Failed to 'git add .gitignore'"
 
-git commit -m "Added .gitignore" || die "Failed to commit .gitignore"
+git commit -m "Adds .gitignore" || die "Failed to commit .gitignore"
```
Comment (jswright61) [inline new-dual-repo.sh:314]: This was drummed into me at a couple of different jobs - The commit comments describe what the commit does to the codebase, not what you did to the codebase that's included in the commit. It's a matter of POV and tense. Very nitpicky.

**Comment by jswright61** [comment]
All of these changes are opinionated (mine) and I'm not trying to hijack your code. I hope you like at least some of the changes. Please pick and choose. I will not be offended if you don't merge any of them.

**Review by tjluoma — APPROVED** [review]
These are good suggestions

**Pull request merged**